### DISCUSSION TITLE
Steuerbare Geräte: Sofortiger Start/Stop

### DISCUSSION THREAD

**Discussion by klaus-schauer** [camueller/SmartApplianceEnabler#467] [Installation und Betrieb]
Falls man bei einem steuerbaren Gerät den sofortigen Start auslöst, wird erwartungsgemäß ein Schaltbefehl per HTTP-Befehl gesendet. Falls man allerdings einige Zeit danach einen sofortigen Stop veranlasst, wird kein HTTP-Schaltbefehl gesendet. Ist das so beabsichtigt?

In welcher Abhängigkeit steht der sofortige Start/Stop mit der Steuerung durch den Home Manager? Wird der Home Manager über den manuellen Start/Stop informiert? Kann der Home Manager weiterhin steuernd eingreifen? Der Status des Verbrauchers wird im konkreten Anwendungsfall per HTTP-Status abgefragt.

**Comment by camueller** [comment]
> Falls man bei einem steuerbaren Gerät den sofortigen Start auslöst, wird erwartungsgemäß ein Schaltbefehl per HTTP-Befehl gesendet. Falls man allerdings einige Zeit danach einen sofortigen Stop veranlasst, wird kein HTTP-Schaltbefehl gesendet. Ist das so beabsichtigt?

Sollte eigentlich funktionieren. Hast Du eine Aktion für "Ausschalten" konfiguriert?

> Wird der Home Manager über den manuellen Start/Stop informiert?

Ja. Dabei wird ihm mitgeteilt, dass für das aktuelle Zeitfenster keine Schaltbefehle akzeptiert werden.

**Reply by klaus-schauer** [reply]
Die Aktion für Ausschalten ist konfiguriert; funktioniert bei einem Ausschaltbefehl durch den Home Manager.

**Comment by klaus-schauer** [comment]
Sorry ich muss mich berichtigen. Das Ausschaltbefehl per HTTP wird gesendet. Er wurde im Zielsystem nicht ausgeführt.

**Reply by klaus-schauer** [reply]
Ich habe mir das noch etwas genauer angesehen:

- Falls der Status auf "grün" steht, so sendet der Ausschaltbefehl auch ein entsprechendes Kommando per HTTP
- Falls der Status aber auf "gelb" gewechselt ist, so wird kein HTTP-Befehl mehr gesendet.

In meinem Fall steuert der Schalter die E-Heizung eines Warmwassererhitzers. Sobald der Thermostat der E-Heizung über die HTTP-Statusabfrage aber ein "ausgeschaltet" sendet, weil die Endtemperatur erreicht ist, geht der Status auf "gelb". Wenn man in dieser Phase aber den Schütz ausschalten will, ist das nicht möglich.

M. E. müsste ein Ausschaltbefehl per HTTP immer gesendet werden, unabhängig davon wie der aktuelle Status des steuerbaren Gerätes ist.

**Comment by camueller** [comment]
Warum solltest Du die etwas ausschalten wollen, für das die HTTP-Statusabfrage ergeben hat, das es bereits ausgeschaltet ist?

Wenn Du Deine Konfiguration und Deine Absicht beschreiben würdest, könnte ich Dir sagen, ob das so mit dem SAE überhaupt funktioniert oder ob das anders gemacht werden sollte.

**Comment by klaus-schauer** [comment]
Per HTTP-Befehl wird die Heizung über einen Schütz an die Versorgungsspannung angeschaltet. Die geheizt wird aber nur, solange die Solltemperatur noch nicht erreicht ist. Das steuert das nachgeschaltete Thermostat der Heizung selbst.

Über die HTTP-Statusabfrage wird aber nur dann "EIN" zurückgemeldet, solange tatsächlich geheizt wird, also das Thermostat den Heizkreis tatsächlich frei gibt. Damit wird dem Home Manager die tatsächliche Einschaltdauer übermittelt. Was für die Berechnung des Verbrauchs notwendig ist.

Nun kann es aber passieren, dass die Solltemperatur erreicht ist, damit der HTTP-Status "AUS" signalisiert wird, aber das Schütz noch eingeschaltet ist. In diesem Status muss es aber dennoch möglich sein, den HTTP-Befehl "AUS" zu senden, sonst kann die Versorgungsspannung per SAE nicht ausgeschaltet werden. Soweit ich das bisher beobachten konnte, funktioniert das solange der Home Manager steuern kann, nur eben nicht bei der manuellen Steuerung über SAE.

**Comment by camueller** [comment]
Du versuchst zwei Geräte in einem Gerät abzubilden. Das merkt man spätestens daran, dass Du etwas ausschalten willst, für das die HTTP-Statusabfrage ergeben hat, dass es bereits ausgeschaltet ist.

Dein Scenarien entspricht dem SG-Ready-Eingang bei Wärmepumpen. Das wird vom SAE über zwei Geräte mit geteiltem Zähler unterstützt: https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/SwitchOnOption_DE.md

**Reply by klaus-schauer** [reply]
Danke für den Tipp. Ich werde mir das mal genauer ansehen. Vielleicht passt das auf meine Anforderungen.

Doch unabhängig davon, wozu gibt es dann den Gerätetyp "Heizung"? Eine Heizung ist nun mal nur thermostatgesteuert und nicht dauerhaft in Betrieb. Mit den Schaltbefehlen des Home Manager gebe ich dann nur dynamisch die Perioden vor, in denen geheizt werden **kann**. Also muss es für den Home Manager  bzw. SAE möglich sein, unabhängig von der Status-Rückmeldung der Heizung EIN- und AUS-Befehle zu senden. Den Status des Schützes und nicht der Heizung zurückzumelden, ist doch sinnlos. Der Schaltzustand des Schützes entspricht doch den SAE-Stati.

**Reply by klaus-schauer** [reply]
Ich möchte die Anfrage nochmals aufgreifen.

Den Tipp mit der Einschaltoption habe ich soweit umgesetzt. Danke dafür.

Dennoch wäre es hilfreich, wenn man den Schaltzustand des zu steuernden Gerätes nicht nur im Status ROT/GRÜN sondern auch im Status GELB manuell steuern könnte und ein entsprechender Schalbefehl z. B. per HTTP gesendet würde.

Ich würde mich freuen, falls dies in der nächsten SAE-Version berücksichtigt werden würde.

**Comment by camueller** [comment]
> Doch unabhängig davon, wozu gibt es dann den Gerätetyp "Heizung"?

Der Gerätetyp ist Bestandteil des SEMP-Protokolls von SMA - alle möglichen Werte wurden von SMA vorgegeben. Wofür dieser Wert verwendet wird, weiss ich nicht. Für den SAE hat dieser Wert keine Relevanz mit Ausnahme des Wertes "Wallbox".
